prow/plugins/blunderbuss/blunderbuss.go

```diff
@@ -222,10 +222,16 @@ func handle(ghc githubClient, roc repoownersClient, log *logrus.Entry, reviewerC
 				if err != nil {
 					return err
 				}
+				var added int
 				combinedReviewers := sets.NewString(reviewers...)
-				combinedReviewers.Insert(approvers...)
-				log.Infof("Added %d approvers as reviewers. %d/%d reviewers found.", combinedReviewers.Len()-len(reviewers), combinedReviewers.Len(), *reviewerCount)
-				reviewers = combinedReviewers.List()
+				for _, approver := range approvers {
+					if !combinedReviewers.Has(approver) {
+						reviewers = append(reviewers, approver)
```

Comment:
Given that the code below does `reviewers = reviewers[:maxReviewers]`  we cannot simply range through approvers and append to this list. There is no randomization happening here, and we need that.

Reply:
indeed, and avoid mixing them when doing so

Reply:
now getReviewers returns a randomized list of approvers/reviewers, but keeping the leaf preference